### ISSUE TITLE
the details

### ISSUE THREAD

**Issue by gandalf-dawn** [structuredlightlab/gradient_ascent_phase_mask_design#2]
ValueError: Unknown mat file type, version 49, 53

errow in python

**Comment by gandalf-dawn** [comment]
Text unknown version

**Comment by hk422** [comment]
Hi @gandalf-dawn,

This refers to the question you asked on this topic in the separate issue [here](https://github.com/structuredlightlab/gradient_ascent_phase_mask_design/issues/1):

```
I can't load the mat.
Can you tell me the environment in which your python and matlab run?
thanks
```

The .mat file was created as a default Matlab's version 7 MAT-file (as using a simple [_save_ function](https://mathworks.com/help/matlab/ref/save.html)). I've just tried loading it using Python on a different machine using exactly the same couple of lines of code appearing in the notebook 

`from scipy.io import loadmat`

`a = loadmat("./Speckle_sorter_bases.mat")["Fibre_speckle_basis"]`

and it worked fine for me both as part of the notebook, or if run separately in the terminal. I can imagine different scipy versions can handle this differently, but "scipy 1.11.3 py311hc1ccb85_0" definitely works without the need of additionally specifying the version of the mat file, etc. You can try it yourself by updating/downgrading your package version:

`pip install scipy==1.11.3`

As a separate note, if you have access to Matlab, you can alternatively load the .mat file into the workspace and save the variables from it in any convenient for you format ('.txt', '.csv', etc.)

Please let me know if this helps or if there are any additional details on this issue.

**Comment by gandalf-dawn** [comment]
Thank you very much for such timely reply，But when I try to load this mat file in matlab, it fails（Matlab's version R2024a）.

![pic2](https://github.com/structuredlightlab/gradient_ascent_phase_mask_design/assets/128908961/ca5679db-a61b-4a43-a4fd-df43fd952fa9)
![pic1](https://github.com/structuredlightlab/gradient_ascent_phase_mask_design/assets/128908961/4cbc6a2b-34ae-4dfd-bd7b-1e9662332fa5)
![pic3](https://github.com/structuredlightlab/gradient_ascent_phase_mask_design/assets/128908961/b2b28917-1617-4d46-bbcd-9912b254e88a)
![pic3](https://github.com/structuredlightlab/gradient_ascent_phase_mask_design/assets/128908961/cab1e6cf-a76c-4876-aaf5-89e68646f01e)

**Comment by gandalf-dawn** [comment]
Is it because I am not using git fls correctly? I'm studying now

**Comment by gandalf-dawn** [comment]
It seems that the content of the mat file cannot be recognized.

**Comment by gandalf-dawn** [comment]
I really appreciate for your help,i have solve the problem.I use git clone to download the map

**Comment by hk422** [comment]
Hi @gandalf-dawn,

It's great to hear you've found the solution to this that works for you. 

Just to additionally comment on the topic of loading the .mat file with Matlab itself: in this case, as the file wasn't saved in an "ASCII" format, a simple `load('Speckle_sorter_bases.mat')` without any additional arguments is enough to load the variables into the workspace. This should work with any Matlab's release. Also, a simple double click on the file itself from the Matlab's Current Folder view would be enough to load the variables into the workspace too, without the need to guess the version of the .mat file. From there, you can always re-save those variables into any convenient format.

![image](https://github.com/structuredlightlab/gradient_ascent_phase_mask_design/assets/64964214/847e978e-9ce1-4206-9dd5-2c1ad14b853e)

![image](https://github.com/structuredlightlab/gradient_ascent_phase_mask_design/assets/64964214/ea36df8a-97e7-45fe-a18c-29f25232bb86)

**Comment by gandalf-dawn** [comment]
thank you！！！



---- Replied Message ----
| From | Hlib ***@***.***> |
| Date | 06/20/2024 17:43 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [structuredlightlab/gradient_ascent_phase_mask_design] the details (Issue #2) |

Closed #2 as completed.

**Comment by gandalf-dawn** [comment]
I am watching the European Cup.



---- Replied Message ----
| From | Hlib ***@***.***> |
| Date | 06/20/2024 17:43 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [structuredlightlab/gradient_ascent_phase_mask_design] the details (Issue #2) |

Closed #2 as completed.
